docs/tasks/administer-cluster/running-cloud-controller.md

```diff
@@ -1,38 +1,96 @@
+* kube-apiserver should not run the Persistent Volume Label admission controller either since the cloud-controller-manager takes over labeling persistent volumes. To prevent the Persistent Volume Label admission plugin from running, make sure the kube-apiserver has a --admission-control flag with a value that does not include PersistentVolumeLabel.
+
+
+As of v1.8, cloud controller manager can implement:
+
+* node controller - responsible for updating nodes using cloud APIs.
+* service controller - responsible for loadbalancers on your cloud against services of type LoadBalancer.
+* route controller - responsible for setting up network routes on your cloud
+* [PersistentVolumeLabel Admission Controller](/docs/admin/admission-controllers/#persistentvolumelabel) - responsible for labeling persistent volumes on your cloud - ensure that the persistent volume label admission plugin is not enabled on your kube-apiserver.
+
+
+## Examples
+
+If you are using a cloud that is currently supported in Kubernetes core and would like to adopt cloud controller manager, see the [cloud controller manager in kubernetes core](https://github.com/kubernetes/kubernetes/tree/master/cmd/cloud-controller-manager).
+
+For cloud controller managers not in Kubernetes core, you can find the respective projects in repos maintained by cloud vendors or sig leads.
+
+* [DigitalOcean](https://github.com/digitalocean/digitalocean-cloud-controller-manager)
+* [Rancher](https://github.com/rancher/rancher-cloud-controller-manager)
+
+For providers already in Kubernetes core, you can run the in-tree cloud controller manager as a [Deployment](/docs/tasks/administer-cluster/cloud-controller-manager-deployment-example.yaml) or [Daemonset](/docs/tasks/administer-cluster/cloud-controller-manager-daemonset-example.yaml) in your cluster.
+
+
+## Limitations
+
+Running cloud controller manager comes with a possible limitations. Although these limitations are being in addressed in upcoming releases, it's important that you are aware of these limitations for production workloads.
+
+### Support for Volumes
+
+Cloud controller manager does not implement any of the volume controllers found in `kube-controller-manager` as volume integrations also require coordination with kubelets. As we evolve CSI (container storage interface) and add stronger support for flex volume plugins, necessary support will be added to cloud controller manager so that clouds can fully integrate with volumes.
+
+### Scalability
+
+In the previous architecture for cloud providers, we relied on kubelets using a local metadata service to retreive node information about itself. With this new architecture, we now fully rely on the cloud controller managers to retrieve information for all nodes. For very larger clusters, you should consider possible bottle necks such as resource requirements and API rate limiting.
```

Comment:
the `--provider-id` flag alleviates this concern. Should we call this out?

Reply:
I think we should have better supporting docs for what the format for --provider-id should be first, what do you think?

Reply:
Sounds good to me.